var/spack/repos/builtin/packages/py-instrain/package.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyInstrain(PythonPackage):
+    """inStrain is python program for analysis of co-occurring genome
+       populations from metagenomes that allows highly accurate genome
+       comparisons, analysis of coverage, microdiversity, and linkage, and
+       sensitive SNP detection with gene localization and synonymous
+       non-synonymous identification."""
+
+    homepage = "https://github.com/MrOlm/instrain"
+    pypi     = "inStrain/inStrain-1.5.7.tar.gz"
+
+    version('1.5.7', sha256='c5dcb01dae244927fe987b5f0695d895ccf521c9dfd87a2cb59057ad50bd9bfa')
+
+    depends_on('python@3.4.0:', type=('build', 'run'))
+    depends_on('py-setuptools@:57', type=('build'))
```

Comment:
Where does the upper bound come from?

Reply:
I don't think so, since I've never used the older version. I'm building it on my local machine with setuptool v59 and seeing if that throws an error. It's possible that I just copy/pasted the dependencies from another package I made and didn't check the version upper/lower bounds that carefully.

Sorry for the sloppiness

Reply:
Can you post the error message here? Wonder what file this is coming from.

Reply:
so I built the package on my local machine with the version restrictions for numba and setuptools removed and it built fine. It seems like those can be removed:

```
Input spec
--------------------------------
py-instrain

Concretized
--------------------------------
py-instrain@1.5.7%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-biopython@1.73%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-numpy@1.21.6%gcc@8.5.0+blas+lapack patches=802970a,873745d arch=linux-fedora36-zen
            ^openblas@0.3.20%gcc@8.5.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=none arch=linux-fedora36-zen
                ^perl@5.34.1%gcc@8.5.0+cpanm+shared+threads arch=linux-fedora36-zen
                    ^berkeley-db@18.1.40%gcc@8.5.0+cxx~docs+stl patches=b231fcc arch=linux-fedora36-zen
                    ^bzip2@1.0.8%gcc@8.5.0~debug~pic+shared arch=linux-fedora36-zen
                        ^diffutils@3.8%gcc@8.5.0 arch=linux-fedora36-zen
                            ^libiconv@1.16%gcc@8.5.0 libs=shared,static arch=linux-fedora36-zen
                    ^gdbm@1.19%gcc@8.5.0 arch=linux-fedora36-zen
                        ^readline@8.1%gcc@8.5.0 arch=linux-fedora36-zen
                            ^ncurses@6.2%gcc@8.5.0~symlinks+termlib abi=none arch=linux-fedora36-zen
                                ^pkgconf@1.8.0%gcc@8.5.0 arch=linux-fedora36-zen
                    ^zlib@1.2.12%gcc@8.5.0+optimize+pic+shared patches=0d38234 arch=linux-fedora36-zen
            ^py-cython@0.29.30%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-pip@21.3.1%gcc@8.5.0 arch=linux-fedora36-zen
                    ^python@3.9.12%gcc@8.5.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-fedora36-zen
                        ^expat@2.4.8%gcc@8.5.0+libbsd arch=linux-fedora36-zen
                            ^libbsd@0.11.5%gcc@8.5.0 arch=linux-fedora36-zen
                                ^libmd@1.0.4%gcc@8.5.0 arch=linux-fedora36-zen
                        ^gettext@0.21%gcc@8.5.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-fedora36-zen
                            ^libxml2@2.9.13%gcc@8.5.0~python arch=linux-fedora36-zen
                                ^xz@5.2.5%gcc@8.5.0~pic libs=shared,static arch=linux-fedora36-zen
                            ^tar@1.34%gcc@8.5.0 zip=pigz arch=linux-fedora36-zen
                                ^pigz@2.7%gcc@8.5.0 arch=linux-fedora36-zen
                                ^zstd@1.5.2%gcc@8.5.0+programs compression=none libs=shared,static arch=linux-fedora36-zen
                        ^libffi@3.4.2%gcc@8.5.0 arch=linux-fedora36-zen
                        ^openssl@1.1.1o%gcc@8.5.0~docs~shared certs=system arch=linux-fedora36-zen
                        ^sqlite@3.38.5%gcc@8.5.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-fedora36-zen
                        ^util-linux-uuid@2.37.4%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-setuptools@59.4.0%gcc@8.5.0 arch=linux-fedora36-zen
                    ^py-wheel@0.37.0%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-h5py@3.6.0%gcc@8.5.0+mpi arch=linux-fedora36-zen
        ^hdf5@1.12.2%gcc@8.5.0~cxx~fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo arch=linux-fedora36-zen
            ^cmake@3.23.1%gcc@8.5.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-fedora36-zen
            ^openmpi@4.1.3%gcc@8.5.0~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~pmi+romio+rsh~singularity+static+vt+wrapper-rpath fabrics=none schedulers=none arch=linux-fedora36-zen
                ^hwloc@2.7.1%gcc@8.5.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl+pci~rocm+shared arch=linux-fedora36-zen
                    ^libpciaccess@0.16%gcc@8.5.0 arch=linux-fedora36-zen
                        ^libtool@2.4.7%gcc@8.5.0 arch=linux-fedora36-zen
                            ^m4@1.4.19%gcc@8.5.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-fedora36-zen
                                ^libsigsegv@2.13%gcc@8.5.0 arch=linux-fedora36-zen
                        ^util-macros@1.19.3%gcc@8.5.0 arch=linux-fedora36-zen
                ^numactl@2.0.14%gcc@8.5.0 patches=4e1d78c,62fc8a8,ff37630 arch=linux-fedora36-zen
                    ^autoconf@2.69%gcc@8.5.0 patches=35c4492,7793209,a49dd5b arch=linux-fedora36-zen
                    ^automake@1.16.5%gcc@8.5.0 arch=linux-fedora36-zen
                ^openssh@9.0p1%gcc@8.5.0 arch=linux-fedora36-zen
                    ^libedit@3.1-20210216%gcc@8.5.0 arch=linux-fedora36-zen
                ^pmix@4.1.2%gcc@8.5.0~docs+pmi_backwards_compatibility~restful arch=linux-fedora36-zen
                    ^libevent@2.1.12%gcc@8.5.0+openssl arch=linux-fedora36-zen
        ^py-mpi4py@3.1.2%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-pkgconfig@1.5.5%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-poetry-core@1.0.8%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-lmfit@1.0.2%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-asteval@0.9.25%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-setuptools-scm@6.3.2%gcc@8.5.0+toml arch=linux-fedora36-zen
                ^py-packaging@21.3%gcc@8.5.0 arch=linux-fedora36-zen
                    ^py-pyparsing@3.0.6%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-tomli@1.2.2%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-scipy@1.8.1%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-pybind11@2.8.1%gcc@8.5.0~ipo build_type=RelWithDebInfo arch=linux-fedora36-zen
                ^ninja@1.10.2%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-pythran@0.11.0%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-beniget@0.4.1%gcc@8.5.0 arch=linux-fedora36-zen
                    ^py-gast@0.5.3%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-ply@3.11%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-uncertainties@3.1.6%gcc@8.5.0~docs~optional arch=linux-fedora36-zen
            ^py-future@0.18.2%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-matplotlib@3.5.2%gcc@8.5.0~animation~fonts+image~latex~movies backend=agg arch=linux-fedora36-zen
        ^freetype@2.11.1%gcc@8.5.0 arch=linux-fedora36-zen
            ^libpng@1.6.37%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-certifi@2021.10.8%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-cycler@0.11.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-fonttools@4.31.2%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-kiwisolver@1.3.2%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-cppy@1.1.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-pillow@9.0.0%gcc@8.5.0~freetype~imagequant+jpeg~jpeg2000~lcms~tiff~webp~webpmux~xcb+zlib arch=linux-fedora36-zen
            ^libjpeg-turbo@2.1.3%gcc@8.5.0 arch=linux-fedora36-zen
                ^nasm@2.15.05%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-python-dateutil@2.8.2%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-six@1.16.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-setuptools-scm-git-archive@1.1%gcc@8.5.0 arch=linux-fedora36-zen
        ^qhull@2020.2%gcc@8.5.0~ipo build_type=RelWithDebInfo arch=linux-fedora36-zen
    ^py-networkx@2.7.1%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-numba@0.55.1%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-llvmlite@0.38.0%gcc@8.5.0 arch=linux-fedora36-zen
            ^binutils@2.38%gcc@8.5.0~gas+gold~headers~interwork+ld~libiberty~lto+nls+plugins libs=shared,static arch=linux-fedora36-zen
            ^llvm@11.1.0%gcc@8.5.0+clang~code_signing+compiler-rt~cuda~flang+gold+internal_unwind~ipo+libcxx~link_llvm_dylib+lld+lldb~llvm_dylib~mlir~omp_debug~omp_tsan+polly~python~shared_libs~split_dwarf~z3 build_type=Release patches=6811ad4,f920173 targets=none version_suffix=none arch=linux-fedora36-zen
                ^perl-data-dumper@2.173%gcc@8.5.0 arch=linux-fedora36-zen
                ^swig@4.0.2%gcc@8.5.0 arch=linux-fedora36-zen
                    ^pcre@8.45%gcc@8.5.0~jit+multibyte+utf arch=linux-fedora36-zen
    ^py-pandas@1.4.2%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-bottleneck@1.3.2%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-numexpr@2.7.3%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-pytz@2021.3%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-psutil@5.8.0%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-pysam@0.18.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^bcftools@1.14%gcc@8.5.0~libgsl~perl-filters arch=linux-fedora36-zen
            ^htslib@1.14%gcc@8.5.0+libcurl+libdeflate arch=linux-fedora36-zen
                ^curl@7.83.0%gcc@8.5.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2~nghttp2 libs=shared,static tls=openssl arch=linux-fedora36-zen
                ^libdeflate@1.10%gcc@8.5.0 arch=linux-fedora36-zen
                    ^gzip@1.12%gcc@8.5.0 arch=linux-fedora36-zen
        ^samtools@1.14%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-pytest@6.2.5%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-attrs@21.4.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-iniconfig@1.1.1%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-pluggy@1.0.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-py@1.11.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-toml@0.10.2%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-scikit-learn@1.1.1%gcc@8.5.0+openmp arch=linux-fedora36-zen
        ^py-joblib@1.1.0%gcc@8.5.0 arch=linux-fedora36-zen
        ^py-threadpoolctl@3.0.0%gcc@8.5.0 arch=linux-fedora36-zen
            ^py-flit@3.6.0%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-docutils@0.18.1%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-flit-core@3.6.0%gcc@8.5.0 arch=linux-fedora36-zen
                ^py-requests@2.26.0%gcc@8.5.0~socks arch=linux-fedora36-zen
                    ^py-charset-normalizer@2.0.12%gcc@8.5.0 arch=linux-fedora36-zen
                    ^py-idna@3.3%gcc@8.5.0 arch=linux-fedora36-zen
                    ^py-urllib3@1.26.6%gcc@8.5.0~brotli~secure~socks arch=linux-fedora36-zen
                ^py-tomli-w@1.0.0%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-seaborn@0.11.2%gcc@8.5.0 arch=linux-fedora36-zen
    ^py-tqdm@4.62.3%gcc@8.5.0~notebook~telegram arch=linux-fedora36-zen

```